tests/test_2772_parquet_extn_array_metadata.py

```diff
@@ -0,0 +1,206 @@
+                ),
+            ),
+        ],
+        None if is_tuple else ["x", "y"],
+        parameters={"foo": "outer"},
+    )
+    for is_tuple in (False, True)
+)
+
+
+@pytest.mark.parametrize(
+    "akarray, as_dict",
+    [
+        (nested_ints, False),
+        (struct_array, False),
+        (with_nulls, False),
+        (option_types, False),
+        (indexed, True),
+        (indexed_tuple, True),
+        (records, True),
+        (records_tuple, True),
+        (indexed, False),
+        (indexed_tuple, False),
+        (records, False),
+        (records_tuple, False),
+    ],
+)
+def test_array_conversions(akarray, as_dict):
+    arrow_natv = ak.to_arrow(
+        akarray, categorical_as_dictionary=as_dict, extensionarray=False
+    )
+    arrow_extn = ak.to_arrow(
+        akarray, categorical_as_dictionary=as_dict, extensionarray=True
+    )
+    extn_field = pa.field("test_field", arrow_extn.type)
+
+    # Convert to native pyarrow types:
+    metadata = collect_ak_arr_type_metadata(extn_field)
+    conv_natv_field = awkward_arrow_field_to_native(extn_field)
+    assert conv_natv_field.type == arrow_natv.type
+    as_natv = array_with_replacement_type(arrow_extn, conv_natv_field.type)
+    assert as_natv == arrow_natv
+
+    # Complete a round-trip, back to AwkwardArrowArray
+    conv_extn_field = native_arrow_field_to_akarraytype(conv_natv_field, metadata)
+    assert conv_extn_field.type == arrow_extn.type
+    assert (
+        conv_extn_field.type._metadata_as_dict() == arrow_extn.type._metadata_as_dict()
+    )
+    conv_metadata = collect_ak_arr_type_metadata(conv_extn_field)
+    assert conv_metadata == metadata
+    # assert_equal_arrowextntypes(conv_extn_field.type, arrow_extn.type)
+    as_extn = array_with_replacement_type(as_natv, conv_extn_field.type)
+    assert as_extn == arrow_extn
+    assert as_extn.type._metadata_as_dict() == arrow_extn.type._metadata_as_dict()
+
+    # And back to Awkward array
+    rt_array = ak.from_arrow(as_extn, highlevel=True)
+    assert to_list(rt_array) == to_list(akarray)
+
+    # Compare awkward types and parameters
+    # Topher 2024-05-23: This is failing on indexed and indexed_tuple,
+    # but it outside the scope of our Arrow table conversions.
+    # I don't *believe* this extra assertion adds any value, but...
+    #
+    # if not isinstance(akarray, ak.Array):
+    #     akarray = ak.Array(akarray)  # A possible issue but this provides .type.show()
+    # ak_type_str_orig = io.StringIO()
+    # ak_type_str_rtrp = io.StringIO()
+    # akarray.type.show(stream=ak_type_str_orig)
+    # rt_array.type.show(stream=ak_type_str_rtrp)
+    # if ak_type_str_orig.getvalue() != ak_type_str_rtrp.getvalue():
+    #     print("  Original type:", ak_type_str_orig.getvalue())
+    #     print("  Rnd-trip type:", ak_type_str_rtrp.getvalue())
+    # assert ak_type_str_orig.getvalue() == ak_type_str_rtrp.getvalue()
```

Comment:
Along with the argument of "this is what git is for" (though it's sometimes hard to find old code in git, especially if it only exists in a PR branch like this), there's also, "this is what `@pytest.mark.skip` is for." You could include this failing test with the `skip` decorator and put the comment about why it's failing in the `reason`.

Then there's the question of "`@pytest.mark.skip`" versus "`@pytest.mark.xfail`". We do expect this to fail, but I don't like the way that pytest shows you the verbose, red stacktrace for tests that are expected to fail—it becomes hard to find the output of tests that are not expected to fail. Or, at least, if `xfail` is going to be used, it should be for tests that will be fixed in the short term, rather than the long term. So I think this one should be `skip`, since I get the impression that it won't be fixed in the short term.

Reply:
I opened issue 3131 and attached a stand-alone unit test there. Your discretion how to resolve this one. It's not clear to me that my expectations are even correct.